### PR TITLE
[WIP] "Linear Assembler"

### PR DESCRIPTION
**Pull request by fredrikekre**
Assemble into `K.nzval` directly.

Missing tests, benchmarks, API, etc etc...

### CODE REVIEW

File: src/assembler.jl [hunk 1]
```diff
@@ -209,3 +209,66 @@ function InsertionSort!(A, order, ii=1, jj=length(A))
...
+        push!(I, d)
+        push!(J, d)
+        push!(V, (d,d))
+    end
+    K = sparse(I, J, V, maximum(I), maximum(J), (x,y) -> x)
+
+    # loop again, and construct the map from (dofi, dofj)
+    # to a linear index into K.nzval
+    dict = Dict{Tuple{Int,Int},Int}()
+    for element_id in 1:ncells
```
Comment (KristofferC) [inline src/assembler.jl]: Why a dict and not a matrix?

Comment (fredrikekre) [reply src/assembler.jl]: Hmm... Interesting. A SparseMatrix that gives the index to another one then? :)

Comment (fredrikekre) [reply src/assembler.jl]: But then this has no advantage over simply `K[i,j] += ...`?

Comment (KristofferC) [reply src/assembler.jl]: You don't have to binary search?

Comment (fredrikekre) [reply src/assembler.jl]: You don't think a hash lookup in a dict is faster than finding the correct index in a SparseMatrixCSC?

File: src/assembler.jl [hunk 2]
```diff
@@ -209,3 +209,67 @@ function InsertionSort!(A, order, ii=1, jj=length(A))
...
+        celldofs!(global_dofs, dh, element_id)
+        @inbounds for j in 1:n, i in 1:n
+            dofi = global_dofs[i]
+            dofj = global_dofs[j]
+            sym && (dofi > dofj && continue)
+            dict[(dofi, dofj)] = findfirst(K.nzval, (dofi, dofj))
```
Comment (fredrikekre) [inline src/assembler.jl]: This line takes forever...

**Comment by fredrikekre** [comment]
This is garbage, right?

**Comment by KristofferC** [comment]
Think so :P 🗑

**Pull request closed without merging**